### PR TITLE
Run flake8 as part of lint script

### PR DESCRIPTION
**Pull request by mkam**
Connected https://github.com/rcbops/u-suk-dev/issues/1537

### CODE REVIEW

**Comment by mkam** [comment]
recheck_lint

File: lint.sh [hunk 1]
```diff
@@ -73,7 +72,7 @@ check_bash(){
 check_python(){
   while read script
   do
-    python -m py_compile  $script \
+    python -m py_compile  $script  && flake8 $script \
```
Comment (hughsaunders) [inline lint.sh]: I don't think both are needed as flake8 will catch compilation errors.

File: Dockerfile [hunk 2]
```diff
@@ -3,7 +3,9 @@ RUN apt-get update && apt-get install -y groovy2 python-pip build-essential pyth
 RUN apt-get install -y libffi-dev
 RUN apt-get install -y sudo
 COPY requirements.txt /requirements.txt
+COPY constraints.txt /test-requirements.txt
```
Comment (hughsaunders) [inline Dockerfile]: I think you meant test-requirements instead of constraints?

**Comment by git-harry** [comment]
@mkam I've added the line
```
Connected https://github.com/rcbops/u-suk-dev/issues/1537
```
to the description so that the pull request and issue are linked in Waffle.

If you add this line to the commit message, the description get populated with it automatically linking the two. In addition, this means the repo history has a record of the issue which can be useful for troubleshooting etc.

**Comment by mkam** [comment]
recheck_lint

File: lint.sh [hunk 3]
```diff
@@ -73,7 +72,7 @@ check_bash(){
 check_python(){
   while read script
   do
-    python -m py_compile  $script \
+    flake8 $script \
       && echo "Python syntax ok: $script" \
       || { echo "Python syntax fail $script"; rc=1; }
   done < <(find ${fargs[@]} -iname \*.py)
```
Comment (mattt416) [inline lint.sh]: Any downside to just running flake8 on the entire dir, rather than on each found py script?

**Comment by mattt416** [comment]
Removing @hughsaunders' review since he's away for some time, plus his concerns have already been addressed.

**Pull request merged**